packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -48,22 +50,68 @@ export class BlockToolAdapter {
    * @param input - input element
    */
   public attachInput(key: DataKey, input: HTMLElement): void {
-    const inputTag = input.tagName as NativeInput;
-
-    /**
-     * @todo Filter non-text-editable inputs
-     */
-    if (![NativeInput.Textarea, NativeInput.Input].includes(inputTag) && !input.isContentEditable) {
-      throw new Error('BlockToolAdapter: input should be either INPUT, TEXTAREA or contenteditable element');
-    }
-
     const caretAdapter = new CaretAdapter(input, this.#model, this.#blockIndex, key);
 
     input.addEventListener('beforeinput', event => this.#handleBeforeInputEvent(event, input, key));
 
     this.#model.addEventListener(EventType.Changed, (event: ModelEvents) => this.#handleModelUpdate(event, input, key, caretAdapter));
   }
 
+  /**
+   * Handles delete events in native input
+   *
+   * @param event - beforeinput event
+   * @param input - input element
+   * @param key - data key input is attached to
+   * @private
+   */
+  #handleDeleteInNativeInput = (event: InputEvent, input: HTMLInputElement | HTMLTextAreaElement, key: DataKey): void => {
+    const inputType = event.inputType as InputType;
+    let start = input.selectionStart as number;
+    const end = input.selectionEnd as number;
+
+    if (start > 0 && start === end) {
+      start = start - 1;
+    }
+    switch (inputType) {
```

Comment:
Please add a comment why this is required
```suggestion
    if (start > 0 && start === end) {
      start = start - 1;
    }

   switch (inputType) {
```

Reply:
Also in case this is a forward delete, we'd need to increase `end` by one